sklearn/utils/tests/test_validation.py

```diff
@@ -324,6 +324,37 @@ def test_check_array_dtype_warning():
     assert_equal(X_checked.format, 'csr')
 
 
+def test_check_array_accept_sparse_type_exception():
+    X = [[1, 2], [3, 4]]
+    X_csr = sp.csr_matrix(X)
+    invalid_type = SVR()
+
+    msg = "A sparse matrix was passed, but dense data is required. " \
+        "Use X.toarray() to convert to a dense numpy array."
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=False)
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=None)
+
+    msg = "Invalid parameter 'accept_sparse={}'"
+    assert_raise_message(ValueError, msg.format([]),
+                         check_array, X_csr, accept_sparse=[])
+    assert_raise_message(ValueError, msg.format(()),
+                         check_array, X_csr, accept_sparse=())
+    assert_raise_message(ValueError, msg.format(invalid_type),
+                         check_array, X_csr, accept_sparse=invalid_type)
+
+    msg = "'SVR' object"
+    assert_raise_message(TypeError, msg,
+                         check_array, X_csr, accept_sparse=[invalid_type])
+
+    # don't raise errors
+    check_array(X_csr, accept_sparse=True)
+    check_array(X_csr, accept_sparse='csr')
+    check_array(X_csr, accept_sparse=['csr'])
+    check_array(X_csr, accept_sparse=('csr'))
```

Comment:
This is not checking what you think it is checking (I guess you meant to test accept_sparse as a tuple) `('csr')` is not a tuple, `('csr',)` is a tuple with one element.

To be honest this four lines need should be moved to another function since they do not test exception. accept_sparse=True, accept_sparse='csr' and accept_sparse=['csr']. It would be great to test the type of the returned object.